bsms.mediawiki

```diff
@@ -0,0 +1,153 @@
+* The Coordinator encrypts this descriptor record with <tt>ENCRYPTION_KEY</tt>.
+* The Coordinator sends the encrypted descriptor record to all participating Signers.
+
+=====Signer=====
+
+* The Signer imports the descriptor record, decrypts it using the <tt>ENCRYPTION_KEY</tt> derived from the open session.
+* The Signer calculates and verifies the descriptor’s <tt>CHECKSUM</tt>. Abort the setup if the <tt>CHECKSUM</tt> is incorrect.
+* The Signer checks whether one of the <tt>KEY</tt>s in the descriptor belongs to it, using path and fingerprint information included in the descriptor. The check must perform an exact match on the <tt>KEY</tt>s, and not using shortcuts such as matching fingerprints (which is trivial to spoof). Abort the setup if it doesn’t detect its own <tt>KEY</tt>.
+* For confirmation, the Signer must display to the user the <tt>CHECKSUM</tt>, plus other configurations, such as <tt>M</tt> and <tt>N</tt>. The total number of Signers, <tt>N</tt>, is important to prevent a <tt>KEY</tt> insertion attack. All participating Signers should be able to display the same confirmation.
+* If all checks pass, the Signer persists the descriptor record in its storage.
+* The Signer can choose to further restrict post-XPUB derivation paths, such as to those defined in [https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki BIP-0044].
+* The Signer should subsequently use the descriptor to generate and verify receive and change addresses.
+
+This completes the setup.
+
+===Encryption===
+
+====The Token====
+We define three modes of encryption.
+
+# <tt>NO_ENCRYPTION</tt> : the <tt>TOKEN</tt> is set to <tt>0</tt>. Encryption is disabled.
+# <tt>STANDARD</tt> : the <tt>TOKEN</tt> is a 64-bit nonce.
+# <tt>EXTENDED</tt> : the <tt>TOKEN</tt> is a 128-bit nonce.
+
+The <tt>TOKEN</tt> can be converted to one of these formats:
+* A mnemonic phrase using [https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki BIP-0039] word list (6 words in <tt>STANDARD</tt> mode, 12 words in <tt>EXTENDED</tt> mode)
+* A decimal number (20 digits in <tt>STANDARD</tt> mode, 40 digits in <tt>EXTENDED</tt> mode)
+* A QR code
+* Other formats
+
+The flexibility in the data format allows each Signer to customize the User Experience based on its respective capabilities.
+
+====Key Derivation====
+The key derivation function is [https://tools.ietf.org/html/rfc2898 PBKDF2], with PRF = SHA512. Specifically:
+
+<tt>DK = PBKDF2(PRF, Password, Salt, c, dkLen)</tt>
+
+Whereas:
+
+* PRF = <tt>SHA512</tt>
+* Password = <tt>"No SPOF"</tt>
+* Salt = <tt>TOKEN</tt>
+* c = <tt>2048</tt>
+* dkLen = <tt>256</tt>
+* DK = Derived <tt>ENCRYPTION_KEY</tt>
+
+====Encryption Scheme====
+The encryption scheme is [https://tools.ietf.org/html/rfc3686 AES, CTR mode].
```

Comment:
yup, will need to flesh out this section further.

Reply:
@dgpv I'm thinking authentication might not be needed because both XPUBs and descriptors have checksums built-in.

Reply:
The signature is only for the signer->coordinator packet. For the coordinator->signer packet, there's only descriptor checksum. The descriptor checksum were not designed as an authentication mechanism. If the cypthertext is modified, the checksum will most likely be incorrect, and even if correct, the descriptor itself most likely be not of the correct format, so on the surface, checksum seems to be enough. But still, this is a guess of a non-cryptographers.

Using signature or checksum+encoding as authentication function and then encrypting would be the 'MAC-then-Encrypt' scheme (https://en.wikipedia.org/wiki/Authenticated_encryption#MAC-then-Encrypt_(MtE)). "Encrypt-then-MAC" seems to be deemed "stronger" than "MAC-then-Encrypt".

Not being a cryptographer, I would choose the way what it is done in some standard, like rfc7366 (https://tools.ietf.org/html/rfc7366#section-3), and then I would not be relying on my guesses.

If we want to avoid extra requirements for explicit MAC, we need to answer the question: is "signature or checksum+encoding as authentication function in the 'MAC-then-Encrypt' scheme" is fine for this purpose and the risk model (where risk model may include a situation when the hww is connected to the host computer under attacker's control). I don't have a knowledge required to answer this.

Reply:
For signer->coordinator, if the coordinator is some sort of online service, the attackers might be able to do many tries while flipping bits in the cyphertext. They would know what part of the xpub or the signature they are targeting, as xpub has fixed size and format and sig has predictable size and format. Since signature has a structure (lengths of s and r components at known positions), this might be more vulnerable to attacks than if a simple hash were used as authentication function.

Reply:
Great points. Let's put out the proposal without authentication and ask for feedback on whether MAC authentication is needed.

Reply:
This was a concern of mine as well, I think it would be good to get feedback on this